File: Linked List/3_Display.js
class Node
{
    constructor(value=0)
    {
        this.data=value
        this.next = null
    }
}

class List
{
    constructor(value=0)
    { 
        const firstNode = new Node(value)
        this.head = firstNode
        this.tail = firstNode
        this.length = 1
    }

    push(value)
    {
        const newNode = new Node(value)
        this.tail.next = newNode
        this.tail = newNode
        newNode.next = null
        this.length++
    }

     /*
    We can display the Linked List by either iterative method or recursive method
    */
    displayIterative()
    {
        /*It is just like displaying an array using loop 
        and printing element at 'i'th index then incrementing i*/
        var current = this.head;
        while(current != null)
        {
            console.log(current.data); 
            current = current.next;
        }
    }

    /**
     * @param current is the node from where values will begin printing
     */
    displayReflexive(current=this.head)
    {
        console.log(current.data);
        if (current.next !=null) 
        this.displayReflexive(current.next);
    }

    /*Here we didn't needed to make two functions for implementing recursive display
    because in JS we can pass the class member as the default value of the argumemnt (this.head in this case)
    for class method*/
}

l = new List(5)
l.push(6)
l.push(7)
l.push(8)
l.displayIterative()
l.displayReflexive()

File: Linked List/LL/3_Display.py
class Node:

    def __init__(self,value=0):
        self.data=value
        self.next = None

class List:

    def __init__(self,value=0):
        
        firstNode = Node(value)
        self.head = firstNode
        self.tail = firstNode
        self.length=1


    def add(self,value):

        newNode = Node(value)
        self.tail.next = newNode
        self.tail = newNode
        newNode.next = None
        self.length+=1


    # We can display the Linked List by either iterative method or recursive method
    def displayIterative(self):

        '''It is just like displaying an array using loop 
        and printing element at 'i'th index then incrementing i'''
        current = self.head
        while(current != None):
            print(current.data)
            current = current.next

    def __recursivePrinter(self,current):
        print(current.data) 
        if current.next != None:
            self.__recursivePrinter(current.next)

    '''
     param start: node from where to begin printing values
     '''
    def displayRecursive(self,start=None):

        if start==None:
             start=self.head; 
        self.__recursivePrinter(start)

l = List(5)
l.add(6)
l.add(7)
l.add(8)
l.displayIterative()
l.displayRecursive()